File: 2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.java
class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        int n = nums.length;
        int[] res = new int[n];
        int st = 0;
        int en = n - 1;

        for (int i = 0, j = n - 1; i < n; i++, j--) {
            if (nums[i] < pivot) {
                res[st++] = nums[i];
            }

            if (nums[j] > pivot) {
                res[en--] = nums[j];
            }
        }

        while (st <= en) {
            res[st++] = pivot;
        }

        return res;
    }
}

File: 2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.py
class Solution:
    def pivotArray(self, nums: List[int], pivot: int) -> List[int]:
        n = len(nums)
        pf = [0] * n
        left, right = 0, n - 1

        for i, j in zip(range(n), range(n - 1, -1, -1)):
            if nums[i] < pivot:
                pf[left] = nums[i]
                left += 1
            if nums[j] > pivot:
                pf[right] = nums[j]
                right -= 1
        
        while left <= right:
            pf[left] = pivot
            left += 1
        
        return pf